### PR TITLE
fix memory leak in HystrixPlugins

### PR DESCRIPTION
**Pull request by chenpengfei**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ *  ] Bug fix
- [ ] New feature provided
- [ ] Improve performance

- Related issues

     #1604
___
### Bug fix
- Bug description.

- How to fix?

     Replace the pr #1605

1. Create SWHystrixPluginsWrapperCache for recording the SWExecutionHookWrapper and SWHystrixConcurrencyStrategyWrapper object. 
2. Store SWHystrixPluginsWrapperCache in EnhancedInstance#dynamicField. 
3. One SWHystrixPluginsWrapperCache attach one EnhancedInstance.

___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

**Comment by chenpengfei** [comment]
yes, close #1605, thank you.

I will send new pull request about dynamicField volatile fix soon.

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/interceptor/enhance/ClassEnhancePluginDefine.java [hunk 1]
```diff
@@ -115,7 +116,7 @@
          *
          */
         if (!context.isObjectExtended()) {
-            newClassBuilder = newClassBuilder.defineField(CONTEXT_ATTR_NAME, Object.class, ACC_PRIVATE)
+            newClassBuilder = newClassBuilder.defineField(CONTEXT_ATTR_NAME, Object.class, ACC_PRIVATE | ACC_VOLATILE)
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/interceptor/enhance/ClassEnhancePluginDefine.java]: This needs to be in another PR.

Comment (chenpengfei) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/interceptor/enhance/ClassEnhancePluginDefine.java]: Define dynamicField volatile flag #1611

File: apm-sniffer/apm-sdk-plugin/hystrix-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/hystrix/v1/HystrixConcurrencyStrategyInterceptor.java [hunk 2]
```diff
@@ -34,8 +34,20 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
...
+        if (wrapperCache == null) {
+            synchronized (objInst) {
+                if (wrapperCache == null) {
+                    wrapperCache = new SWHystrixPluginsWrapperCache();
+                    objInst.setSkyWalkingDynamicField(wrapperCache);
+                }
+            }
+        }
+
+        wrapperCache.getSwHystrixConcurrencyStrategyWrapper().compareAndSet(null, new SWHystrixConcurrencyStrategyWrapper((HystrixConcurrencyStrategy) ret));
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/hystrix-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/hystrix/v1/HystrixConcurrencyStrategyInterceptor.java]: 1. This compareAndSet executes every time, I have a performance concern because of `AtomicReference` includes lock.
2. The `swHystrixConcurrencyStrategyWrapper` could be inited inside double-check, no `AtomicReference` required. Only a `volatile` key word should be enough.

**Comment by wu-sheng** [comment]
I set `requested changes` just for avoiding someone merges this :) 

@chenpengfei Let's talk and test about it, then we decide how to continue, thank you for finding and fixing this bug.
